src/display/annotation_layer.js

```diff
@@ -858,14 +846,37 @@ class TextWidgetAnnotationElement extends WidgetAnnotationElement {
    *
    * @private
    * @param {HTMLDivElement} element
-   * @param {Object} font
    * @memberof TextWidgetAnnotationElement
    */
-  _setTextStyle(element, font) {
-    // TODO: This duplicates some of the logic in CanvasGraphics.setFont().
+  _setTextStyle(element) {
+    const TEXT_ALIGNMENT = ["left", "center", "right"];
+    const { fontName, fontSize, fontColor } = this.data.defaultAppearanceData;
     const style = element.style;
-    style.fontSize = `${this.data.fontSize}px`;
-    style.direction = this.data.fontDirection < 0 ? "rtl" : "ltr";
+
+    /**
+     *  TODO: If font size is 0, calculate the font-size based on the height and
+     *          width of the element
+     *  For now, just use the default font size
+     */
+    if (fontSize) {
+      style.fontSize = `${fontSize}px`;
+    }
+
+    style.color = `rgb(${fontColor[0]}, ${fontColor[1]}, ${fontColor[2]})`;
+
+    if (this.data.textAlignment !== null) {
+      element.style.textAlign = TEXT_ALIGNMENT[this.data.textAlignment];
+    }
+
+    let font = null;
+    if (
+      fontName &&
+      fontName.name &&
+      this.data.fontRefName &&
```

Comment:
I think this line can be removed because of https://github.com/mozilla/pdf.js/issues/12226#issuecomment-682163009. As far as I could tell, but please check again, nothing actually sets this `fontRefName` variable, so it should be dead code. I believe `fontName.name` that is used here is the replacement for this.

Reply:
This whole block is dead code, and furthermore there's no way that this actually works (it looks untested):
 - First of all, the `data.fontRefName` property is never actually set.
 - Secondly, the `fontName` cannot be used to lookup a font given that we index fonts using an *internal* PDF.js-specific name rather than what exists in the `/DA` of the annotation.

All-in-all, this code-block should probably just be remove since there's no way that it works as-is (and getting it to work would require a lot more changes and testing).

Reply:
I agree, I reused this block from what was there. I did test it with other fonts, but it did not work as expected, so I will remove handling the font name.